behave_django/management/commands/behave.py

```diff
@@ -98,22 +123,118 @@ def add_behave_arguments(parser):  # noqa
             keywords['help'] = keywords['config_help']
             del keywords['config_help']
 
-        parser.add_argument(*option_strings, **keywords)
+        try:
+            parser.add_argument(*option_strings, **keywords)
+        except ArgumentError:
+            remove_option(parser, *fixed)
+            parser.add_argument(*option_strings, **keywords)
 
 
 class Command(BaseCommand):
     help = 'Runs behave tests'
+    test_runner = None
 
-    def add_arguments(self, parser):
+    DEFAULT_TEST_RUNNER = 'behave_django.runner.BehaviorDrivenTestRunner'
+    SIMPLE_TEST_RUNNER = 'behave_django.runner.SimpleTestRunner'
+    EXISTING_TEST_RUNNER = 'behave_django.runner.ExistingDatabaseTestRunner'
+
+    def init_test_runner(self, argv=None):
+        """
+        Pre-parse the command line to extract the value of the --testrunner
+        option. This allows a test runner to define additional command line
+        arguments.
+        """
+
+        argv = argv or sys.argv
+        self.test_runner = get_command_line_option(argv, '--testrunner')
+        use_simple = get_command_line_option(argv, '-S', '--simple',
+                                             action='store_true',
+                                             default=False)
+        use_existing_database = get_command_line_option(
+            argv,
+            '--use-existing-database',
+            action='store_true',
+            default=False
+        )
+        use_dry_run = get_command_line_option(
+            argv,
+            '--dry-run',
+            action='store_true',
+            default=False
+        )
+        if use_simple:
+            # TODO: Delete in upcoming version
+            self.stderr.write(self.style.WARNING(
+                '-S/--simple has been depricated, please use: '
+                '"--testrunner behave_django.runner.SimpleTestRunner"'
+            ))
+            self.test_runner = self.SIMPLE_TEST_RUNNER
+        elif use_existing_database:
+            # TODO: Delete in upcoming version
+            self.stderr.write(self.style.WARNING(
+                '--use-existing-database has been depricated, please use: '
+                '"--testrunner behave_django.runner.'
+                'ExistingDatabaseTestRunner"'
+            ))
+            self.test_runner = self.EXISTING_TEST_RUNNER
+        elif use_dry_run:
+            # TODO: Delete in upcoming version
```

Comment:
Same here: TODO will trigger a violation of the linter. Add a deprecation notice instead.